memgpt/local_llm/vllm/api.py

```diff
@@ -0,0 +1,51 @@
+import os
+from urllib.parse import urljoin
+import requests
+
+from ..utils import load_grammar_file, count_tokens
+
+WEBUI_API_SUFFIX = "/generate"
+DEBUG = False
+
+
+def get_vllm_completion(endpoint, model, prompt, context_window, settings={}, grammar=None):
+    """https://github.com/vllm-project/vllm/blob/main/examples/api_client.py"""
+    prompt_tokens = count_tokens(prompt)
+    if prompt_tokens > context_window:
+        raise Exception(f"Request exceeds maximum context length ({prompt_tokens} > {context_window} tokens)")
+
+    # Settings for the generation, includes the prompt + stop tokens, max length, etc
+    request = settings
```

Comment:
Do we also need to pass context length? Or is that done when the server is started (on the server-side)?

Reply:
@cpacker The context length is set when you start the server so you do not need to pass it as an argument